_posts/2024-01-19-trino-2023-wrapped.md

```diff
@@ -0,0 +1,253 @@
+the [launch of the Trino Gateway]({{site.url}}{% post_url
+2023-09-28-trino-gateway %}). A proxy and
+load-balancer, it’s a crucial piece of Trino infrastructure for organizations
+that need more than one Trino cluster to suit their needs.
+
+Why would you want more than one Trino cluster? Maybe you want one cluster with
+fault-tolerant execution enabled for ETL workloads and another cluster for
+speedy ad-hoc analytics. Perhaps you have analysts performing wildly
+differently-sized queries, and high-volume compute-intensive queries are proving
+to be bad neighbors for lightweight and low-latency queries that shouldn’t take
+more than milliseconds. Historically, you would have to manually manage swapping
+between clusters, establish a new connection, and try not to get a headache in
+the process.
+
+Enter the Trino Gateway! By routing all of your Trino traffic automatically,
+it’s never been easier to manage, maintain, and query multiple Trino clusters at
+once. Load balancing ensures that no one cluster gets overworked, and it’s the
+perfect way to stop large queries from getting in the way of the little guys.
+Add in the fact that you can seamlessly shut down an individual cluster for
+updates or maintenance while the Trino Gateway routes traffic elsewhere, and
+it’s easy to see why this is such a game-changer. We’re super excited for it to
+be out there in the world, and we hope it makes running Trino at the largest
+scales simpler and faster than ever before.
+
+For more information on the Trino Gateway, check out:
+
+* [The announcement blog post]({{site.url}}{% post_url 2023-09-28-trino-gateway %})
+* [The quickstart guide](https://github.com/trinodb/trino-gateway/blob/main/docs/quickstart.md)
+* [The main Trino Gateway repo](https://github.com/trinodb/trino-gateway/tree/main)
+
+## New features!
+
+### SQL Routines (User-defined functions!)
+
+Whether you want to refer to them as SQL routines or as user-defined functions,
+they’re a big deal. Fresh off the presses and only a few months old, they do
+exactly what you’d expect them to do: you, a user, can define and re-use your
+own functions! Define and use them inline as part of a query to make that query
+cleaner, easier, and simpler to understand. Or, if you’re really cooking, you
+can run a query that defines the routine on the schema or catalog, allowing
+other Trino users querying it to access the same function time and time again as
+part of their other queries. It’s a level of customization that we’ve never had
+before in Trino, and no longer do you need to write your own Java plugins in
+order to create and re-use functions that do exactly what you need them to do.
+
+If you want to learn more about SQL routines, you can check
+out [the introduction to SQL routines](/docs/current/routines/introduction.html)
+in our documentation, as well as a
+few [example routines](/docs/current/routines/examples.html) which give a good
+look at how they can be used.
```

Comment:
Also demo in SQL training

Reply:
we should work the training in anyway with link to the four videos .. maybe do that here

Reply:
Here the link to the SQL training video - https://www.youtube.com/watch?v=1siAYR6BzzY&list=PLFnr63che7wYzZoo5yyEF5R1QrOH6VRq3&index=4